src/http.ts

```diff
@@ -42,46 +42,52 @@ export type DevicesResponse = { devices: DevicesByName } | ErrorResponse;
 export type QMIResponse = { qmi: QMI } | ErrorResponse;
 export type QMIsResponse = { qmis: QMI[] } | ErrorResponse;
 
-export async function _request(path: string, data: any = {}, method: string) {
+type PromiseRequestCallback = (ok: (value?: any) => void, err: (value?: any) => void) => request.RequestCallback;
+const defaultHandler: PromiseRequestCallback = (ok, err) => (e, r, b) => {
```

Comment:
Given that this is getting a bit complex it would benefit from a doctoring explaining how it's used elsewhere in the source: aka who typically calls the handler with `(ok, err)`, and how the resulting function with args `(e, r, b)` are ultimately consumed